mathesar_ui/src/systems/table-view/table-inspector/record/RowActions.svelte

```diff
@@ -28,22 +32,46 @@
       }
     }
   }
+
+  $: ({ savedRecords } = recordsData);
+  $: ({ columns } = columnsDataStore);
+
+  function getRecord(selectedRowIndex: number) {
+    return $savedRecords[selectedRowIndex].record;
+  }
```

Comment:
Open the Records tab, and try creating a new record. The uncaught error breaks the entire product.

This is an unsafe access, where saved records may not have the row index when new records are present. There is a method called `getRecordRows` in records.ts, which should be used for any record access. Undefined return should be assumed on any index access on arrays, where length check isn't done.

Reply:
> Undefined return should be assumed on any index access on arrays, where length check isn't done.

This has been bugging me for a while, and I've opened #1966 in search of a more systemic solution.